cocos/2d/components/rich-text.ts

```diff
@@ -611,12 +611,16 @@ export class RichText extends Component {
         let curEnd = longStr.length / 2;
         let curString = longStr.substring(curStart, curEnd);
         let leftString = longStr.substring(curEnd);
-        let curStringSize = this._calculateSize(styleIndex, curString);
-        let leftStringSize = this._calculateSize(styleIndex, leftString);
+        const curStringSize = this._calculateSize(styleIndex, curString).clone();
+        const leftStringSize = this._calculateSize(styleIndex, leftString).clone();
+        let maxWidth = this._maxWidth;
+        if (this._maxWidth === 0) {
```

Comment:
Why needs this?

Reply:
If the user does not enter maxWidth, the original algorithm does not take this case into account, resulting in incorrect typesetting